server/src/main/java/io/spine/server/commandbus/CommandBus.java

```diff
@@ -352,24 +328,31 @@ protected void checkFieldsSet() {
         @CheckReturnValue
         public CommandBus build() {
             checkFieldsSet();
+            commandScheduler =
+                    ServerEnvironment.instance()
+                                     .newCommandScheduler();
+            flowWatcher = createFlowWatcher();
+            commandScheduler.setWatcher(flowWatcher);
 
-            if (commandScheduler == null) {
-                commandScheduler = new ExecutorCommandScheduler();
-            }
-            flowWatcher = new CommandFlowWatcher((tenantId) -> {
-                @SuppressWarnings("OptionalGetWithoutIsPresent") // ensured by checkFieldsSet()
-                SystemWriteSide writeSide = system().get();
-                SystemWriteSide result = delegatingTo(writeSide).get(tenantId);
-                return result;
-            });
-            commandScheduler.setFlowWatcher(flowWatcher);
+            TenantIndex tenantIndex = tenantIndex().orElseThrow(tenantIndexNotSet());
+            tenantConsumer = tenantIndex::keep;
 
             CommandBus commandBus = createCommandBus();
+
             commandScheduler.setCommandBus(commandBus);
 
             return commandBus;
         }
 
+        private CommandFlowWatcher createFlowWatcher() {
+            return new CommandFlowWatcher((tenantId) -> {
+                @SuppressWarnings("OptionalGetWithoutIsPresent") // ensured by checkFieldsSet()
```

Comment:
Now that this code is extracted into a separate method, I wouldn't rely on `checkFieldsSet()`.